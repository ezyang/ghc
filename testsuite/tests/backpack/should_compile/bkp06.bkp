package p where
    signature A1 where
        data A = A { foo :: Int, bar :: Bool }
    signature A2 where
        data A = A { foo :: Int, bar :: Bool }
    signature A2(foo) where
        import A1(foo)
