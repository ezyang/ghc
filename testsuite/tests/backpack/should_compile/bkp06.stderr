[1 of 1] Processing p
/--- Shape for p
provides:
requires: A1 -> p(hole:A1, hole:A2):A1
                A{A, A, foo, bar}
          A2 -> p(hole:A1, hole:A2):A2, p(hole:A1, hole:A2):A2
                hole:A1.A{hole:A1.A, hole:A1.A, hole:A1.bar, hole:A1.foo}
signature hole:A1
signature hole:A2
signature hole:A2
\---
  [1 of 3] Compiling A1[abstract sig] ( p/A1.hsig, nothing )
  [2 of 3] Compiling A2[abstract sig] ( p/A2.hsig, nothing )
  [3 of 3] Compiling A2[abstract sig] ( p/A2.hsig, nothing )
